### ISSUE TITLE
Help with sentiment prediction with trained imdb keras model

### ISSUE THREAD

**Issue by dping1** [fchollet/keras#7026]
Hi, 

Can someone please help provide some guidances on how to prepare input data to make predictions using a trained model with IMDB dataset?   Below are what I have tried:

I used the Keras example below and trained a model for sentiment analysis.  The accuracy was 86%.  

https://github.com/Theo-/sentiment-analysis-keras-conv

I assumed the integers in the IMDB training/testing dataset were just indexes from the IMDB vocabulary, 
so I wrote some code to make sentiment prediction for some new texts using the steps below.
1) load the word index using keras.dataset.get_word_index
2) replace all words in the new text with indexes from the IMDB word index, and generate an ndarray.  I also pad the ndarray with a maxlen of 1600 to match maxlen used in training and skip any word not found in the vocabulary.  
3) call model.predict() to generate a prediction using the ndarray from step 2 above.

As the trained model had a 86% accuracy, I was expecting good prediction accuracy for the new texts, however, I found the prediction accuracy to be poor.  

I figured it could be the texts I wrote were not good quality, so I downloaded the original IMDB text data from the link below, and tested it again with clearly marked pos/neg sample texts, and results were still not accurate.

http://ai.stanford.edu/~amaas/data/sentiment/aclImdb_v1.tar.gz

However, if I just test with samples from the IMDB testing dataset, the results are very accurate.  So apparently the way I prepare the input data is not correct.

Below are my questions:
1)  How is the keras IMDB dataset prepared?
2)  Is there any way to restore the original text from the training/testing dataset?  Apparently, the training data/testing datasets are different from the original text.  I tried to convert the indexes to original words, and the outputs do not even look like to proper English.
3)  What are the right steps to prepare input data for making a prediction with IMDB?

Thanks,
David

**Comment by jorgecarleitao** [comment]
I believe that you would get an answer faster by posting this question in StackOverflow or related. This place is about issues of Keras itself. If you do that, please close this issue :)
